### ISSUE TITLE
Assertion `var->value->type == init_value->value.type' failed.  with nested struct @typeInfo

### ISSUE THREAD

**Issue by bheads** [ziglang/zig#1091]
Getting an error when trying to read nested struct typeInfo:

zig: /home/bheads/zig/src/codegen.cpp:2813: LLVMOpaqueValue* ir_render_decl_var(CodeGen*, IrExecutable*, IrInstructionDeclVar*): Assertion `var->value->type == init_value->value.type' failed.

```C
     const Network = struct {
         host: []const u8,
         port: u16,
     };
 
     const Foo = struct {
         a: u32,
         network : Network, // remove this and it works
         b: i32,
     };
 
     inline for (@typeInfo(Foo).Struct.fields) |f| {
         const name = f.name;
         const ftype = @typeName(f.field_type);
         warn("field {} {} \n", name, ftype);
    }
```

**Comment by dbandstra** [comment]
I ran into this too, here's my code 

```
const builtin = @import("builtin");

fn find(comptime T: type) ?*T {
  return null;
}

fn generic(comptime SelfType: type) void {
  inline for (@typeInfo(SelfType).Struct.fields) |field| {
    comptime const ComponentType = @typeInfo(field.field_type).Pointer.child;

    // this line seems to be causing the crash
    const maybe_component = find(ComponentType);

    // it works fine if i use this in an expression. only assigning it to a
    // variable fails
    // if (find(ComponentType) != null) {}
  }
}

const A = struct{ unused: u32 };
const B = struct{ unused: u32 };

const Data = struct{
  // works if both fields are the same time!
  one: *A,
  two: *B,
};

test "" {
  generic(Data);
}
```

```
zig: /home/dbandstra/zig/src/codegen.cpp:2878: LLVMOpaqueValue* ir_render_decl_var(CodeGen*, IrExecutable*, IrInstructionDeclVar*): Assertion `var->value->type == init_value->value.type' failed.
```

**Comment by dbandstra** [comment]
#845 #740 #741 #917

**Comment by andrewrk** [comment]
fixed by 09cc1dc66067f378a1508e34c0714b659b445724
